x-pack/plugins/security/server/config.ts

```diff
@@ -82,11 +86,23 @@ export function createConfig$(context: PluginInitializerContext, isTLSEnabled: b
         secureCookies = true;
       }
 
-      return {
+      // "sessionTimeout" is deprecated and replaced with "session.idleTimeout"
+      // however, NP does not yet have a mechanism to automatically rename deprecated keys
+      // for the time being, we'll do it manually:
+      const sess = config.session;
+      const session = {
+        idleTimeout: (sess && sess.idleTimeout) || config.sessionTimeout || null,
+        lifespan: (sess && sess.lifespan) || null,
+      };
+
+      const val = {
         ...config,
         encryptionKey,
         secureCookies,
+        session,
       };
+      delete val.sessionTimeout; // DEPRECATED
+      return val;
```

Comment:
This is ugly and I hate it, but it appears necessary until the New Platform can provide full support for deprecated config settings.